### PR TITLE
refactor passport

### PR DESCRIPTION
**Pull request by sylvainlap**
refactor passport-local, no more session

### CODE REVIEW

File: files/api/user/services/passport.js [hunk 1]
```diff
@@ -167,31 +167,22 @@ passport.callback = function * (ctx, next) {
...
-        let user = yield passport.protocols.local.login(ctx);
-        next(null, user);
-      } catch (err) {
-        next(err);
-      }
-    } else {
-      next(new Error('Invalid action'));
+  if (provider === 'local' && action === 'register') {
+    try {
+      let user = yield passport.protocols.local.register(ctx);
```
Comment (Aurelsicoko) [inline files/api/user/services/passport.js]: I believe you can use a constant variable ;)

File: files/api/user/services/passport.js [hunk 2]
```diff
@@ -167,31 +167,22 @@ passport.callback = function * (ctx, next) {
...
+      next(null, user);
+    } catch (err) {
+      next(err);
     }
-  } else {
+  } else if (action === 'connect') {
     yield passport.authenticate(ctx.params.provider, function * (err, user) {
```
Comment (Aurelsicoko) [inline files/api/user/services/passport.js]: Why do you need to call this function as a generator?

File: files/api/user/services/protocols/local.js [hunk 3]
```diff
@@ -91,14 +91,12 @@ exports.register = function * (ctx) {
...
  */
 
-exports.login = function * (ctx) {
-  const deferred = Promise.defer();
-
-  // Format params and init variables.
-  const params = _.merge({}, ctx.params, ctx.request.body, ctx.request.query);
-  const identifier = params.identifier;
+exports.login = function (ctx, identifier, password, next) {
   let query = {};
```
Comment (Aurelsicoko) [inline files/api/user/services/protocols/local.js]: I believe you can use a constant variable too ;)

File: files/api/user/services/passport.js (hunk 1 above)
Comment (sylvainlap) [inline files/api/user/services/passport.js]: okay

File: files/api/user/services/passport.js (hunk 2 above)
Comment (sylvainlap) [inline files/api/user/services/passport.js]: The koa-passport authenticate callback has to be a generator function.

See https://github.com/rkusa/koa-passport/blob/master/lib/framework/koa.js#L78

File: files/api/user/services/protocols/local.js (hunk 3 above)
Comment (sylvainlap) [inline files/api/user/services/protocols/local.js]: okay

File: files/api/user/services/passport.js (hunk 2 above)
Comment (Aurelsicoko) [inline files/api/user/services/passport.js]: Okay, my bad ;)

**Pull request merged**